Port hardshrink_backward and softshrink_backward to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58661 Port silu_backward to structured
* #58665 Port gelu_backward to structured
* #58664 Port gelu to structured
* #58663 Port hardshrink to structured
* **#58662 Port hardshrink_backward and softshrink_backward to structured**



Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

Freey0 committed:
Port hardshrink_backward and softshrink_backward to structured

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -123,6 +123,13 @@ TORCH_META_FUNC(hardsigmoid_backward) (const Tensor& grad_output, const Tensor&
+TORCH_META_FUNC(hardshrink_backward) (
+  const Tensor & grad, const Tensor & self, const Scalar& lambd
+) {
+  set_output(0, self.sizes(), {}, self.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT), {});
```
Hmm, this is weird......

ezyang replied:
Yeah, I don't think this line is necessary

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).

Freey0 committed:
Update on "Port hardshrink_backward and softshrink_backward to structured"


Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

[ghstack-poisoned]

Freey0 committed:
Update on "Port hardshrink_backward and softshrink_backward to structured"




Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

[ghstack-poisoned]

Freey0 committed:
Update on "Port hardshrink_backward and softshrink_backward to structured"




Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).

Freey0 committed:
Update on "Port hardshrink_backward and softshrink_backward to structured"




Differential Revision: [D28572532](https://our.internmc.facebook.com/intern/diff/D28572532)

[ghstack-poisoned]

Freey0 commented:
Test failure is not relevant

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572532).
